sys/newlib_syscalls_default/syscalls.c

```diff
@@ -56,12 +56,67 @@
 #include "xtimer.h"
 #endif
 
+#ifndef NUM_HEAPS
+#define NUM_HEAPS 1
+#endif
+
 /**
  * @brief manage the heap
  */
 extern char _sheap;                 /* start of the heap */
 extern char _eheap;                 /* end of the heap */
-char *heap_top = &_sheap + 4;
+
+extern char _sheap1;
+extern char _eheap1;
+
+extern char _sheap2;
+extern char _eheap2;
+
+extern char _sheap3;
+extern char _eheap3;
+
+struct heap {
+    char* start;
+    char* end;
+};
+
+static char *heap_top[NUM_HEAPS] = {
+    &_sheap,
+#if NUM_HEAPS > 1
+    &_sheap1,
+#endif
+#if NUM_HEAPS > 2
+    &_sheap2,
+#endif
+#if NUM_HEAPS > 3
+    &_sheap3,
+#endif
```

Comment:
How about ?
```suggestion
#else
#error Unsupported NUM_HEAPS value
#endif
```

Reply:
Then it would also error if `NUM_HEAPS <= 3`, but I added a warning if the number is still bigger.